sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -27,12 +25,12 @@ import org.apache.commons.lang3.StringUtils
  */
 object CodeFormatter {
   def format(code: CodeAndComment): String = {
-    new CodeFormatter().addLines(
-      StringUtils.replaceEach(
-        code.body,
-        code.comment.keys.toArray,
-        code.comment.values.toArray)
-    ).result
+    val formatter = new CodeFormatter
+    code.body.split("\n").foreach { line =>
+      val trimmed = line.trim
+      formatter.addLine(code.comment.getOrElse(trimmed, trimmed))
+    }
+    formatter.result()
```

Comment:
cc @sarutak , here I assume the placeholder will always take a entire line, is it corrected?